Readme.md

```diff
@@ -1,22 +1,83 @@
 
-<p>In some cases, when the default <a href="http://help.devexpress.com/#AspNet/CustomDocument3753"><u>filter row</u></a> editor's functionality is not enough, you can provide custom filter cell content using the <a href="http://help.devexpress.com/#AspNet/DevExpressWebASPxGridViewGridViewColumn_FilterTemplatetopic"><u>FilterTemplate</u></a>. </p><p>In this example, a default cell editor is replaced with the <a href="http://documentation.devexpress.com/#AspNet/clsDevExpressWebASPxGridLookupASPxGridLookuptopic"><u>ASPxGridLookup</u></a> control. The control's <a href="http://documentation.devexpress.com/#AspNet/DevExpressWebASPxEditorsScriptsASPxClientEdit_ValueChangedtopic"><u>ValueChanged</u></a> client-side event is used to send a callback to the server side, invoking the grid's <a href="http://help.devexpress.com/#AspNet/DevExpressWebASPxGridViewASPxGridView_CustomCallbacktopic"><u>CustomCallback</u></a> event. In the event handler, a filter criteria is created and applied to the grid using the <a href="http://help.devexpress.com/#AspNet/DevExpressWebASPxGridViewASPxGridView_ApplyFilterToColumntopic"><u>ApplyFilterToColumn</u></a> method.</p>
+    ```aspx
+    <dx:GridViewDataTextColumn FieldName="CategoryName">
+        <FilterTemplate>
+            <dx:ASPxGridLookup runat="server" ID="Lookup" AutoGenerateColumns="False" KeyFieldName="CategoryID"
+                DataSourceID="CategoriesDataSource" SelectionMode="Multiple" TextFormatString="{0}">
+                <Columns>
+                    <dx:GridViewCommandColumn ShowSelectCheckbox="true" />
+                    <dx:GridViewDataTextColumn FieldName="CategoryName" />
+                    <dx:GridViewDataBinaryImageColumn FieldName="Picture">
+                        <PropertiesBinaryImage ImageWidth="48" />
+                    </dx:GridViewDataBinaryImageColumn>
+                </Columns>
+                <ClientSideEvents ValueChanged="Lookup_ValueChanged" />
+            </dx:ASPxGridLookup>
+        </FilterTemplate>
+    </dx:GridViewDataTextColumn>
+    ```
 
-<br/>
+3. Handle the editor's client-side [ValueChanged](https://docs.devexpress.com/AspNet/js-ASPxClientEdit.ValueChanged) event. In the handler, call the grid's client-side `PerformCallback` method to send a callback to the server.
+
+    ```js
+    function Lookup_ValueChanged(s, e) {
+        grid.PerformCallback("FilterByCategories");
+    }
+    ```
+
+4. Handle the grid's server-side `CustomCallback` event. In the handler, create a custom filter criteria based on the edit value and call the grid's [ApplyFilterToColumn](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.ApplyFilterToColumn(DevExpress.Web.GridViewDataColumn-DevExpress.Data.Filtering.CriteriaOperator)) method to apply this filter to the grid.
```

Comment:
Someone would say it's a questionable matter but I can't help but notice this -- "criterion" is a Greek word where "criterion" is singular, and "criteria" is a plural form. Therefore, if you say "criteria" (in plural), you don't need an article.

```suggestion
4. Handle the grid's server-side `CustomCallback` event. In the handler, create custom filter criteria based on the edit value and call the grid's [ApplyFilterToColumn](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.ApplyFilterToColumn(DevExpress.Web.GridViewDataColumn-DevExpress.Data.Filtering.CriteriaOperator)) method to apply the filter criteria to the grid.
```